src/shared/switch-root.c

```diff
@@ -81,7 +81,7 @@ int switch_root(const char *new_root,
                 /* When the path was not a mount point, then we need to reopen the path, otherwise, it still
                  * points to the underlying directory. */
 
-                fd = open(new_root, O_DIRECTORY|O_CLOEXEC);
+                fd = openat(new_root_fd, ".", O_DIRECTORY|O_CLOEXEC);
```

Comment:
Neither this nor `open(FORMAT_PROC_FD_PATH(new_root_fd), O_DIRECTORY|O_CLOEXEC)` do not work.
Still opened file descriptor points to the underlying directory, if my understanding is correct. Interesting.